### DISCUSSION TITLE
Add Easy Configuration Option for Customizing Favicon in Swagger UI and ReDoc

### DISCUSSION THREAD

**Discussion by nh916** [fastapi/fastapi#11632] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
something like we are already doing with the docs logo


def create_custom_openapi(app: FastAPI) -> Callable[[], dict[str, Any]]:
    def custom_openapi() -> dict[str, Any]:
        if app.openapi_schema:
            return app.openapi_schema

        title: str = "My API"
        version: str = "1.0.0"
        summary: str = "API documentation"
        description: str = "my description"
        logo_url: str = "https://my-website-assets/my-logo"
        docs_favicon_url: str = "https://my-website-assets/my-favicon"    # request for this

        openapi_schema = get_openapi(
            title=title,
            version=version,
            summary=summary,
            description=description,
            routes=app.routes,
        )

        openapi_schema["info"]["x-logo"] = {"url": logo_url}
        openapi_schema["info"]["docs-logo"] = {"url": docs_favicon_url}    # request for this

        app.openapi_schema = openapi_schema
        return app.openapi_schema

    return custom_openapi
```

### Description

There is currently no straightforward way to customize the favicon for the Swagger UI and ReDoc documentation in FastAPI through configuration settings and customizing the HTML code is more effort than it needs to be I think.

I think if we can offer a way to customize the favicon the same way we do for the logo, that could be really easy and neat for the developers using the software

### Operating System

Linux, Windows, macOS, Other

### FastAPI Version

0.109.2

### Pydantic Version

2.6.0

### Python Version

Python 3.12.1

**Comment by tiangolo** [comment, marked as answer]
Thanks for the interest! There are many things that could be customizable, e.g. the source of the JS files, and many other things. I prefer to keep it simple and have a simple default that works for most cases. And for the cases where someone would want some extra flexibility, they can override the entire docs, including all the possible parameters we could imagine, with the current approach. So I would rather keep it as it is. 🤓

**Comment by nh916** [comment]
okay cool

thank you so much!
